test/guardianManager.js

```diff
@@ -143,7 +140,7 @@ describe("GuardianManager", function () {
                 }
             });
 
-            it("should add many Guardians (relayed transaction)", async () => {
+            it("should add many Guardians", async () => {
```

Comment:
Why not name tests in a way that distinguishes relayed from non-relayed transactions?

Reply:
It's obvious from the test itself, if a `manager.relay` call is there then it's _relayed_. When searching for relayed transactions that's the string I'll search for, rather than the name of the test. The name should be a representation of the condition being tested rather than details of its execution. This is also prone to errors as there's no check the title matches the execution.

Reply:
> The name should be a representation of the condition being tested rather than details of its execution.

I think a test name should ideally contain no less information than "when in state S0, performing action A yields state S1" (unless obvious from context). Here the two tests (relayed and not relayed) share the same initial and expected final states but differ by the action being tested. Even though we expect those two actions to always be equivalent, they are still distinct actions performed by a user (which is why we write a different test for each). I'm not sure what you mean by 'details of its execution'.

> This is also prone to errors as there's no check the title matches the execution.

Isn't this the case with any test? How does specifying the action being performed make it more error prone?

Also, more pragmatically, etherlime test does not always return the line on which the test failed, so being able to search the test suite using a test name that is more specific saves a bit of time.

Reply:
Perhaps this needs a larger discussion. I have started a [list of items to discuss around unit testing](https://trello.com/c/h7wl8ohs) and will setup a meeting towards the end of this sprint to discuss and agree what if anything needs changing. For now I'll revert back the test names above.